src/main/java/org/osc/controller/nsc/restserver/api/InspectionApis.java

```diff
@@ -0,0 +1,403 @@
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.log4j.Logger;
+import org.osc.controller.nsc.api.SampleSdnRedirectionApi;
+import org.osc.controller.nsc.entities.InspectionHookEntity;
+import org.osc.controller.nsc.entities.InspectionPortEntity;
+import org.osc.controller.nsc.entities.NetworkElementEntity;
+import org.osc.controller.nsc.utils.RedirectionApiUtils;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.jdbc.DataSourceFactory;
+import org.osgi.service.jpa.EntityManagerFactoryBuilder;
+import org.osgi.service.transaction.control.TransactionControl;
+import org.osgi.service.transaction.control.jpa.JPAEntityManagerProviderFactory;
+
+@Component(service = InspectionApis.class)
+@Path("/nw")
+@Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+public class InspectionApis {
+    private static Logger LOG = Logger.getLogger(InspectionApis.class);
+    private EntityManager em;
+    private RedirectionApiUtils utils;
+    private SampleSdnRedirectionApi api;
+    @Reference(target = "(osgi.local.enabled=true)")
+    private TransactionControl txControl;
+    @Reference(target = "(osgi.unit.name=nsc-mgr)")
+    private EntityManagerFactoryBuilder builder;
+    @Reference(target = "(osgi.jdbc.driver.class=org.h2.Driver)")
+    private DataSourceFactory jdbcFactory;
+    @Reference(target = "(osgi.local.enabled=true)")
+    private JPAEntityManagerProviderFactory resourceFactory;
+    private String ipAddr;
+
+    public void init(String ipAddr) throws Exception {
+        if (ipAddr != null) {
+            if (ipAddr.equals(this.ipAddr)) {
+                return;
+            }
+        } else {
+            return;
+        }
+
+        Properties props = new Properties();
+        props.setProperty(JDBC_URL,
+                "jdbc:h2:./nscPlugin_" + ipAddr + ";MVCC\\=TRUE;LOCK_TIMEOUT\\=10000;MV_STORE=FALSE;");
+        props.setProperty(JDBC_USER, "admin");
+        props.setProperty(JDBC_PASSWORD, "admin123");
+        DataSource ds = null;
+        try {
+            ds = this.jdbcFactory.createDataSource(props);
+        } catch (SQLException error) {
+            LOG.error(error);
+            throw new IllegalStateException(error.getMessage(), error);
+        }
+        EntityManager em = this.resourceFactory
+                .getProviderFor(this.builder, singletonMap("javax.persistence.nonJtaDataSource", (Object) ds), null)
+                .getResource(this.txControl);
+        this.em = em;
+        this.api = new SampleSdnRedirectionApi(this.txControl, this.em);
+        this.utils = new RedirectionApiUtils(this.em, this.txControl);
+        this.ipAddr = ipAddr;
+
+    }
+
+    @Path("/ipAddr/{ipAddr}/netelem")
```

Comment:
ipAddr -> controller/{id} . I understand we are using the ip address to identify. However we should keep the API more generic, the ip here is just being used as an id for the controller.

Reply:
ok, will take care.